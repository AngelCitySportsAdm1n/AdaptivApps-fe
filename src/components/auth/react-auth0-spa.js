import React, { useState, useEffect, useContext } from 'react';
import createAuth0Client from '@auth0/auth0-spa-js';
import PropTypes from 'prop-types';

const DEFAULT_REDIRECT_CALLBACK = () =>
  window.history.replaceState({}, document.title, window.location.pathname);

export const Auth0Context = React.createContext();
export const useAuth0 = () => useContext(Auth0Context);
//console.log('console.log useAuth0', useAuth0.getTokenSilently);
export const Auth0Provider = ({
  children,
  onRedirectCallback = DEFAULT_REDIRECT_CALLBACK,
  ...initOptions
}) => {
  const [isAuthenticated, setIsAuthenticated] = useState();
  const [user, setUser] = useState();
  const [auth0Client, setAuth0] = useState();
  const [loading, setLoading] = useState(true);
  const [popupOpen, setPopupOpen] = useState(false);

  useEffect(() => {
    const initAuth0 = async () => {
      const auth0FromHook = await createAuth0Client(initOptions);
      setAuth0(auth0FromHook);

      if (
        window.location.search.includes('code=') &&
        window.location.search.includes('state=')
      ) {
        const { appState } = await auth0FromHook.handleRedirectCallback();
        onRedirectCallback(appState);
      }
      const tokenP = auth0FromHook.getIdTokenClaims();

      const isAuthenticated = await auth0FromHook.isAuthenticated();
      //console.log(
      //'Showing this auth0FrameHook',
      //   auth0FromHook.getIdTokenClaims()
      // );
      // checks to make sure user is authenticated and sets user if true.
      setIsAuthenticated(isAuthenticated);

      if (isAuthenticated) {
        const token = await tokenP;
        //console.log('Showing tokenP', token.__raw);
        //localStorage.setItem('golden', token.__raw);
        const user = await auth0FromHook.getUser();
        setUser(user);
        //console.log('authenticated', user);
      }

      setLoading(false);
    };
    initAuth0();
    // eslint-disable-next-line
  }, []);

  const loginWithPopup = async (params = {}) => {
    setPopupOpen(true);
    try {
      await auth0Client.loginWithPopup(params);
    } catch (error) {
      //console.error(error);
    } finally {
      setPopupOpen(false);
    }
    const user = await auth0Client.getUser();
    setUser(user);
    setIsAuthenticated(true);
    //console.log('user', user);
  };

  const handleRedirectCallback = async () => {
    setLoading(true);
    await auth0Client.handleRedirectCallback();
    const user = await auth0Client.getUser();
    setLoading(false);
    setIsAuthenticated(true);
    setUser(user);
    //console.log('redirect', user);
  };
  return (
    <Auth0Context.Provider
      value={{
        isAuthenticated,
        user,
        loading,
        popupOpen,
        loginWithPopup,
        handleRedirectCallback,
        getIdTokenClaims: (...p) => auth0Client.getIdTokenClaims(...p),
        loginWithRedirect: (...p) => auth0Client.loginWithRedirect(...p),
        getTokenSilently: (...p) => auth0Client.getTokenSilently(...p),
        getTokenWithPopup: (...p) => auth0Client.getTokenWithPopup(...p),
        logout: (...p) => auth0Client.logout(...p),
      }}
    >
      {children}
    </Auth0Context.Provider>
  );
};

Auth0Provider.propTypes = {
  children: PropTypes.any,
  onRedirectCallback: PropTypes.any,
};
