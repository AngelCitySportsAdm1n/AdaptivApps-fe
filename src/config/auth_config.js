module.exports = {
  domain: process.env.REACT_APP_CLIENT_DOMAIN,
  clientId: process.env.REACT_APP_CLIENT_ID,
  roleUrl: 'http://adaptivapps.com/roles',
  audience: process.env.REACT_APP_AUDIENCE,
  responseType: 'token id_token',
  scope: 'openid email',
};
