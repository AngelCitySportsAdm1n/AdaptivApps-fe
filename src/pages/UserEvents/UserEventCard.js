//React imports
import React from 'react';
//Component imports
import { Link } from '@reach/router';
// GraphQL/Apollo imports
import { useMutation } from 'react-apollo';
import { UNREGISTER_FROM_EVENT } from './queries';
// Auth0 imports
import { useAuth0 } from '../../config/react-auth0-spa';
//Styling imports
import { Flex, Container, Button } from 'adaptiv-ui';
import PropTypes from 'prop-types';


export default function UserEventCard({ event, refetch }) {
  const [updateProfile] = useMutation(UNREGISTER_FROM_EVENT);
  // Retrieves current user info
  const { user } = useAuth0();
  // Unregisters user from specified event
  const unregisterFromEvent = () => {
    updateProfile({
      variables: { id: event.id, email: user.email }
    });
    refetch();
  };

  return (
    <Flex col>
      <Container bg_src={event.imgUrl} h="10vh" w="20rem"></Container>
      <small>
        {event.startDate} - {event.endDate}
      </small>
      <h6>
        <b>{event.title}</b>
      </h6>
      <p>location</p>
      <Button secondary="true" onClick={unregisterFromEvent}>Unregister</Button>
      <Link to={`${event?.id}`}>View Details</Link>
    </Flex>
  );
}

UserEventCard.propTypes = {
  event: PropTypes.object,
};