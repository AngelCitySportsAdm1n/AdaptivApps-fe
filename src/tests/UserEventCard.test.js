import React from 'react';
import UserEventCard from '../pages/UserEvents/UserEventCard';
import renderer from 'react-test-renderer';
import { ApolloProvider } from '@apollo/react-hooks';
import ApolloClient from 'apollo-boost';
import { getIdTokenClaims } from '../config/react-auth0-spa';

const client = new ApolloClient({
  uri: process.env.REACT_APP_API_URL,
  credentials: 'same-origin',
  request: async operation => {
    const token = await getIdTokenClaims();
    // Attach token to headers
    operation.setContext(context => ({
      headers: {
        ...context.headers,
        Authorization: token.__raw,
      },
    }));
  },
});

it('renders user event card correctly', () => {
  const event = {
    id: 12345,
    title: "Test Event",
    startDate: "3/25/20",
    endDate: "3/26/20",
    location: "Test Location",
    activities : [
      {
        id: 12345,
        name: "Test Activity",
        startDate: "3/25/20",
        location: "Test Location",
        startTime: "8:00AM"
      }
    ]
  };
  
  const tree = renderer
    .create(
      <ApolloProvider client={client} >
        <UserEventCard event={event} />
      </ApolloProvider>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});