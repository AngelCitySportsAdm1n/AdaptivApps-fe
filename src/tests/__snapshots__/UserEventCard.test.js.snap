// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user event card correctly 1`] = `
<div
  className="sc-ifAKCX jKRaVs"
>
  <div
    className="sc-gZMcBi iauavI"
  />
  <small>
    3/25/20
     - 
    3/26/20
  </small>
  <h6>
    <b>
      Test Event
    </b>
  </h6>
  <p>
    Test Location
  </p>
</div>
`;
